tests/EdFi.Admin.DataAccess.IntegrationTests/Repositories/PostgreSQL/ClientAppRepoTests.cs

```diff
@@ -0,0 +1,727 @@
+            {
+                var tmpClient = _clientAppRepo.GetClient(_testClient.Key, null);
+                tmpClient.ShouldBeNull();
+            }
+        }
+
+        [TestFixture]
+        public class When_getting_a_client_and_all_related_properties : ClientAppRepoTests
+        {
+            private const string ClientName = "zsdfasdf";
+            private ApiClient _result;
+
+            [SetUp]
+            public void SetUp()
+            {
+                // Arrange
+                // Adding vendor, namespace prefix, profiles, education organizations,
+                // and ownership token in order to test the Entity Framework
+                // hydration of those properties on the client
+                var client = new ApiClient
+                {
+                    Name = ClientName,
+                    Key = "987sal;ifasd",
+                    Secret = "987asdkif"
+                };
+                OpenContext.Clients.Add(client);
+
+                var vendor = new Vendor
+                {
+                    VendorName = "vendor name"
+                };
+                OpenContext.Vendors.Add(vendor);
+
+                var namespacePrefix = new VendorNamespacePrefix
+                {
+                    NamespacePrefix = "abc",
+                    Vendor = vendor
+                };
+                vendor.VendorNamespacePrefixes.Add(namespacePrefix);
+                OpenContext.VendorNamespacePrefixes.Add(namespacePrefix);
+
+                var application = new Application
+                {
+                    Vendor = vendor,
+                    ApplicationName = "a",
+                    OperationalContextUri = "uri://ed-fi.org"
+                };
+                client.Application = application;
+                application.ApiClients.Add(client);
+
+                vendor.Applications.Add(application);
+                OpenContext.Applications.Add(application);
+
+                var profile = new Profile
+                {
+                    ProfileName = "profile"
+                };
+                application.Profiles.Add(profile);
+                profile.Applications.Add(application);
+                OpenContext.Profiles.Add(profile);
+
+                var token = new OwnershipToken
+                {
+                    Description = "sddsdsf"
+                };
+                token.ApiClients.Add(client);
+                client.CreatorOwnershipToken = token;
+                OpenContext.OwnershipTokens.Add(token);
+
+                OpenContext.SaveChangesForTest();
+
+                // Uncomment next two lines only for debugging.
+                // PostgreSQL does not have a "read uncommitted" isolation level, therefore
+                // this is the only way to manually query the database for inspection while
+                // debugging the test.
+                //Transaction.Complete();
+                //Transaction = new TransactionScope();
```

Comment:
This test class has some failing tests. All of the other tests are passing on my system. The lines above may be helpful in trying to debug the issue.

Reply:
All tests pass on [target branch ](https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS/tree/ODS-6250)